modules/tensor_mechanics/src/materials/ComputeFiniteStrain.C

```diff
@@ -174,3 +128,91 @@ ComputeFiniteStrain::computeQpStrain()
   _mechanical_strain[_qp] = _rotation_increment[_qp] * _mechanical_strain[_qp] * _rotation_increment[_qp].transpose();
   _total_strain[_qp] = _rotation_increment[_qp] * _total_strain[_qp] * _rotation_increment[_qp].transpose();
 }
+
+void
+ComputeFiniteStrain::computeQpIncrements(RankTwoTensor & total_strain_increment, RankTwoTensor & rotation_increment)
+{
+  if (_decomposition_method == "Rashid")
```

Comment:
1. switch is faster.
2. switch is much clearer (readability and easy maintenance).